pytorch_lightning/trainer/trainer.py

```diff
@@ -1681,6 +1681,8 @@ def _call_callbacks_on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> Dict
         callback_states = {}
         for callback in self.callbacks:
             # TODO: Add profiling for on_save_checkpoint hook
+            # FIXME: the error raised is because this manual call does not set `_current_fx_name`
+            # we should also do `_call_callback_hook(...)` but need the output state
             state = callback.on_save_checkpoint(self, self.lightning_module, checkpoint)
```

Comment:
@daniellepintz the error you see appears because this call needs to be updated too, otherwise the `_current_fx_name` does not get set properly.

If you need help ping me on Slack!

Reply:
Thanks for your help Carlos! I updated the code to set `_current_fx_name` here. I don't think there is a way to do it by calling `_call_callback_hook` since we have special logic for getting and setting `callback_states` in these two methods (`_call_callbacks_on_save_checkpoint`/`_call_callbacks_on_load_checkpoint`)